runtime-transaction/src/compute_budget_instruction_details.rs

```diff
@@ -23,7 +24,12 @@ pub(crate) struct ComputeBudgetInstructionDetails {
     requested_compute_unit_price: Option<(u8, u64)>,
     requested_heap_size: Option<(u8, u32)>,
     requested_loaded_accounts_data_size_limit: Option<(u8, u32)>,
-    num_non_compute_budget_instructions: u32,
+    // instructions counts
+    num_compute_budget_instructions: u32,
+    num_builtin_instructions: u32,
+    num_non_builtin_instructions: u32,
+    // A list of migration feature IDs for builtin instructions
+    maybe_builtin: Vec<Pubkey>,
```

Comment:
we can't allocate like this per transaction

Reply:
This is proof of concept, to wire up cached "maybe builtins" to static meta, implement the functions.

To improve performance, maybe need @buffalojoec [PR](https://github.com/anza-xyz/agave/pull/3790#discussion_r1860013703), then `Vec<pubkey>` can be replaced with `maybe_builtin: [u12; 10]` - a fixed array that maps `BUILTINS`, each `u12` counts number of builtin instructions. That way `sanitize` can refer to `BUILTINS` for migration feature id.

Reply:
91929894ef4061cf2f7dfdaaa7a858226cebb345 optimized `Vec<Pubkey>` away, replaced with fixed-length vector of counter that has the same size of static migration feature IDs, it helps a bit. The main cost introduced is hashing pubkey for lookup at `BUILTIN_INSTRUCTION_COSTS`

Added a benchmark with transaction has 3 migrating builtins (stake, config and atl), the results are:

<details><summary>at a3a0a5904dba9994d70ad8d9da4ad4458fe0aa01 , basically master</summary>

```
bench_process_compute_budget_instructions_with_migrating_builtins/3 migrating builtins
                        time:   [14.361 µs 14.396 µs 14.433 µs]
                        thrpt:  [70.948 Melem/s 71.133 Melem/s 71.302 Melem/s]
```

</details>

<details><summary>at a5a4b63878b462a402af5773076a8ac3f0d0b207, early version of using Vec<Pubkey></summary>

```
bench_process_compute_budget_instructions_with_migrating_builtins, SIMD-170 activated: true/3 migrat...
                        time:   [63.799 µs 63.939 µs 64.115 µs]
                        thrpt:  [15.971 Melem/s 16.015 Melem/s 16.050 Melem/s]
                 change:
                        time:   [-1.9105% -1.6424% -1.3531%] (p = 0.00 < 0.05)
                        thrpt:  [+1.3717% +1.6699% +1.9477%]
                        Performance has improved.
Found 8 outliers among 100 measurements (8.00%)
  2 (2.00%) low severe
  2 (2.00%) high mild
  4 (4.00%) high severe

Benchmarking bench_process_compute_budget_instructions_with_migrating_builtins, SIMD-170 activated: false/3 migra...: Collecting 100 samples in estimated 5.2009 s (81k iterations
bench_process_compute_budget_instructions_with_migrating_builtins, SIMD-170 activated: false/3 migra...
                        time:   [64.422 µs 64.547 µs 64.761 µs]
                        thrpt:  [15.812 Melem/s 15.865 Melem/s 15.895 Melem/s]
                 change:
                        time:   [-0.1473% +0.0274% +0.2268%] (p = 0.77 > 0.05)
                        thrpt:  [-0.2263% -0.0274% +0.1475%]
                        No change in performance detected.
Found 4 outliers among 100 measurements (4.00%)
  1 (1.00%) low mild
  1 (1.00%) high mild
  2 (2.00%) high severe
```

</details>

<details><summary>at 91929894ef4061cf2f7dfdaaa7a858226cebb345, top of this branch </summary>

```
bench_process_compute_budget_instructions_with_migrating_builtins, SIMD-170 activated: true/3 migrat...
                        time:   [59.712 µs 59.759 µs 59.812 µs]
                        thrpt:  [17.120 Melem/s 17.135 Melem/s 17.149 Melem/s]
                 change:
                        time:   [+24.631% +24.895% +25.221%] (p = 0.00 < 0.05)
                        thrpt:  [-20.141% -19.933% -19.763%]
                        Performance has regressed.
Found 4 outliers among 100 measurements (4.00%)
  2 (2.00%) high mild
  2 (2.00%) high severe

Benchmarking bench_process_compute_budget_instructions_with_migrating_builtins, SIMD-170 activated: false/3 migra...: Collecting 100 samples in estimated 5.1244 s (86k iterations
bench_process_compute_budget_instructions_with_migrating_builtins, SIMD-170 activated: false/3 migra...
                        time:   [59.735 µs 59.877 µs 60.142 µs]
                        thrpt:  [17.026 Melem/s 17.102 Melem/s 17.142 Melem/s]
                 change:
                        time:   [+23.871% +24.792% +26.899%] (p = 0.00 < 0.05)
                        thrpt:  [-21.197% -19.867% -19.271%]
                        Performance has regressed.
Found 4 outliers among 100 measurements (4.00%)
  4 (4.00%) high severe
```

</details>